website/docs/advanced-concepts/advanced-integrations.md

```diff
@@ -0,0 +1,133 @@
+---
+sidebar_position: 0
+---
+
+# Integration - Advanced
+
+This section goes into detail on how [Integrations](../concepts/integration/) work. If you are just looking to use Functionless without extending it, this section can be skipped. If you have not read [Integrations](../concepts/integration/), do that first.
+
+## Terms
+
+- Resource - Infrastructure and Business Logic which use an integration, generally modelled as some infrastructure like a [`Function`](../concepts/function/) or [`EventBus`](../concepts/event-bridge/).
+- [Integration](../concepts/integration/) - An object (may be a Resource) or function which can be used within a Resource.
+- [Integration Type](./integration-types.md) - A pattern that an integration implements to be used within one or more Resources.
+
+```mermaid
+graph LR;
+  IntegrationType(Integration Type)
+  Resource--uses-->Integration;
+  Resource--supports-->IntegrationType;
+  Integration--implements-->IntegrationType;
+```
+
+A view of what a user sees of Integrations and Resources:
+
+```ts
+// event bus is a Resource
+const bus = new EventBus<Event<{ name: string }>>(this, "bus");
+
+// StepFunction is a Resource
+const sfn = new StepFunction(this, sfn, (payload: { name: string }) => {
+  // bus.putEvents is an Integration that supports the StepFunction ASL Integration Type
+  bus.putEvents({
+    source: "specialSource",
+    "detail-type": "UserNameEvent",
+    detail: payload,
+  });
+});
+```
+
+Partial view of `EventBus` and `putEvents` which supports the StepFunction Integration Type:
+
+```ts
+class EventBus {
+
+  public readonly putEvents: IntegrationCall<
+    "EventBus.putEvents",
+    IEventBus<E>["putEvents"]
+  >;
+
+  constructor(...) {
+    // the makeIntegration method helps generate the right object and type for all integrations
+    // also makes the property callable
+    this.putEvents = makeIntegration<"EventBus.putEvents", IEventBus<E>["putEvents"]>({
+      // ASL in an IntegrationType that requests a method of the form (call: CallExpr, context: ASL) => Omit<State, "Next">;
+      asl: (call: CallExpr, context: ASL) => {
+        // do things like grant permissions, generate ASL, and output a State object.
+      }
+    })
+  }
```

Comment:
I would attempt to separate out each of the points written in comments into paragraphs and code snippets.  When writing docs, try and step through each piece very slowly.